homeassistant/components/shelly/entity.py

```diff
@@ -31,22 +33,39 @@ def shelly_naming(self, block, entity_type: str):
             and self.wrapper.device.settings["mode"] == "roller"
         ):
             channels = 1
-
     entity_name = self.wrapper.name
+    if not block:
+        return f"{entity_name} {self.description.name}"
     if channels > 1 and block.type != "device":
         entity_name = self.wrapper.device.settings["relays"][int(block.channel)]["name"]
         if not entity_name:
             entity_name = f"{self.wrapper.name} channel {int(block.channel)+1}"
-
     if entity_type == "switch":
         return entity_name
-
     if entity_type == "sensor":
         return f"{entity_name} {self.description.name}"
-
     raise ValueError
 
 
+def shelly_rest_parser(self, path: str):
```

Comment:
Don't use `self` as a name in a standalone function. If you want it to be a class method , you should but it in the class definition.

Reply:
I did the same for naming and it was merged; I'm a bit lost

Reply:
shelly naming should be fixed, it is not a class method